lib/Test/MockTime/HiRes.pm

```diff
@@ -80,7 +80,7 @@ sub gettimeofday() {
     if (defined $Test::MockTime::fixed) {
         return wantarray ? do {
             my $int_part = int($Test::MockTime::fixed);
-            ($int_part, 1_000_000 * ($Test::MockTime::fixed - $int_part))
+            ($int_part, 1_000_000 * sprintf('%.6f', ($Test::MockTime::fixed - $int_part)))
```

Comment:
I see, then, my real question was "why it is not `round(1_000_000 * ($Test::MockTime::fixed - $int_part))` where `round` is from [`Math::Round`](http://search.cpan.org/dist/Math-Round/Round.pm)?"  I'm worrying that the floating point number returned by `sprintf` may be converted internally during the operation of `*` to a native floating point number which cannot be expressed in 6 digits.  I guess this doesn't happen for a floating point number represented in a string but I'm not sure.  So please clarify this.